### PR TITLE
Seperate the blockexplore into a seperate siae binary

### PR DESCRIPTION
**Pull request by triazo**
Branch is on top of my other pull request.

### CODE REVIEW

**Comment by DavidVorick** [comment]
need to resolve the merge conflicts

File: siae/daemon_test.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+// TestStartDaemon probes the startDaemon function.
+func TestStartDaemon(t *testing.T) {
+	testDir := build.TempDir("siad", "TestStartDaemon")
+	config.Siad.NoBootstrap = false
+	config.Siad.APIaddr = "localhost:45170"
+	config.Siad.RPCaddr = ":45171"
+	config.Siad.HostAddr = ":45172"
+	config.Siad.SiaDir = testDir
```
Comment (DavidVorick) [inline siae/daemon_test.go]: @lukechampine: how do we deal with port testing again?

I think you can just use ":0" for all of them

File: siae/main.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		Short: "Print version information",
+		Long:  "Print version information about the Sia Daemon",
+		Run:   versionCmd,
+	})
+
+	// Set default values, which have the lowest priority.
+	root.PersistentFlags().BoolVarP(&config.Siad.NoBootstrap, "no-bootstrap", "n", false, "disable bootstrapping on this run")
+	root.PersistentFlags().StringVarP(&config.Siad.APIaddr, "api-addr", "a", "localhost:9980", "which host:port the API server listens on")
+	root.PersistentFlags().StringVarP(&config.Siad.RPCaddr, "rpc-addr", "r", ":9981", "which port the gateway listens on")
+	root.PersistentFlags().StringVarP(&config.Siad.SiaDir, "sia-directory", "d", "", "location of the sia directory")
```
Comment (DavidVorick) [inline siae/main.go]: Seems odd to me to duplicate so much code between siad and siae. I wonder if they should be the same binary.

File: siae/main_test.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	// Try running and closing an instance of siad.
+	os.Args = []string{
+		"siad",
+		"-n",
+		"-a",
+		"localhost:45150",
+		"-r",
+		"localhost:45151",
+		"-H",
+		"localhost:45152",
```
Comment (DavidVorick) [inline siae/main_test.go]: ":0" is probably okay.

**Comment by triazo** [comment]
It does seem a bit odd to have two top level binaries that are so similar. Due to the import tree, you don't even save in size of the binary by splitting the blockexplorer out. Alternatively we could have a command line flag to put it into blockexplorer mode, or generalize that to flags that toggle every module from the default. Also, we could make a hardlink to the binary and switch based on arg0. The amount of things we need to switch is small and I think there are numerous ways to prevent code replication.

File: siae/daemon_test.go (hunk 1 above)
Comment (lukechampine) [inline siae/daemon_test.go]: you should be able to use ":0", but you'll need to call various methods to discover which port the OS has assigned you. See other testing code for examples

**Comment by DavidVorick** [comment]
I would like to merge this. We can postpone the discussion about whether or not 'siae' should exist until we have an alternative (probably a stack of flags in 'siad' that enables and disables all of the modules).

**Comment by triazo** [comment]
Updated and rebased. Should merge now.

File: siae/main_test.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	testDir := build.TempDir("siad", "TestMain")
+
+	// Try running and closing an instance of siad.
+	os.Args = []string{
+		"siad",
+		"-n",
+		"-a",
+		"localhost:45150",
+		"-r",
+		"localhost:45151",
```
Comment (DavidVorick) [inline siae/main_test.go]: these should be :0 I believe

**Pull request merged**